pkg/reader/usdc_reader.go

```diff
@@ -212,6 +230,12 @@ func AllAvailableDomains() map[uint64]uint32 {
 	return destDomains
 }
 
+type evmUSDCMessageReader struct {
```

Comment:
minor, but if you extracted Solana to a separate file then I would extract evm implementation to `usd_reader_evm.go` and keep only the factory/interfaces in `usdc_reader.go`